client/flutter/lib/api/content_loading.dart

```diff
@@ -105,3 +111,48 @@ class ContentLoading {
     return '${name}.$suffix.yaml';
   }
 }
+
+class WhoCacheManager extends BaseCacheManager {
+  static const key = "whoCache";
+
+  static WhoCacheManager _instance;
+
+  // Attempting to change the maxAgeCacheObject parameter here has no effect
+  // on http content, for which the manager uses the cache control max-age header.
+  WhoCacheManager._init() : super(key);
+
+  factory WhoCacheManager() {
+    if (_instance == null) {
+      _instance = WhoCacheManager._init();
+    }
+    return _instance;
+  }
+
+  // This changes the behavior of the base method to wait for an update of the
+  // expired file before returning, rather than favoring the cached version.
+  @override
+  Future<File> getSingleFile(String url, {Map<String, String> headers}) async {
+    var cacheFile = await getFileFromCache(url);
+    if (cacheFile != null) {
+      if (cacheFile.validTill.isBefore(DateTime.now())) {
+        // Wait for the refreshed file.
+        await webHelper.downloadFile(url, authHeaders: headers);
```

Comment:
I think we have a different requirement than normal HTTP caches.
If this call fails, we should still give the stale cached file.  A cached file is still better for the user than the baked in file, for WHO content files.

Reply:
good point... fixing.

Reply:
I've changed this.